pkg/ccl/changefeedccl/changefeed_dist.go

```diff
@@ -461,6 +461,10 @@ func makePlan(
 		if execCtx.ExecCfg().Settings.Version.IsActive(ctx, clusterversion.V25_4) {
 			progressConfig = &execinfrapb.ChangefeedProgressConfig{
 				PerTableTracking: changefeedbase.TrackPerTableProgress.Get(sv),
+				// If the per table pts flag was turned on between changefeed creation and now,
+				// the per table pts records will be rewritten in the new format when the
+				// highwater mark is updated in manageProtectedTimestamps.
+				PerTableProtectedTimestamps: changefeedbase.PerTableProtectedTimestamps.Get(sv),
```

Comment:
As I mentioned in https://github.com/cockroachdb/cockroach/pull/149616#discussion_r2274260510, we should only consider enabling this if `PerTableTracking` is true